workers/unity/Packages/io.improbable.gdk.testutils/Editor/SpatialdManager.cs

```diff
@@ -0,0 +1,297 @@
+            }
+
+            var deploymentData = (List<object>) deploymentsObj;
+
+            return deploymentData
+                .OfType<Dictionary<string, object>>()
+                .Select(data =>
+                {
+                    var id = (string) data["id"];
+                    var name = (string) data["name"];
+
+                    if (!data.TryGetValue("tag", out var tagsObj))
+                    {
+                        tagsObj = new List<object>();
+                    }
+
+                    var tags = (List<object>) tagsObj;
+                    return new LocalDeployment(this, id, name, ProjectName, tags.Cast<string>().ToArray());
+                })
+                .ToList();
+        }
+
+        public void Dispose()
+        {
+            var result = RedirectedProcess.Command(Common.SpatialBinary)
+                .WithArgs("service", "stop")
+                .InDirectory(Common.SpatialProjectRootDir)
+                .RedirectOutputOptions(OutputRedirectBehaviour.None)
+                .RunAsync()
+                .Result;
+
+            if (result.ExitCode != 0)
+            {
+                Debug.LogWarning($"Failed to stop spatiald with error: {string.Join("\n", result.Stderr)}");
+            }
+        }
+
+        /// <summary>
+        ///     Represents a local deployment.
+        /// </summary>
+        public struct LocalDeployment : IDisposable
+        {
+            /// <summary>
+            ///     The ID of this deployment.
+            /// </summary>
+            public readonly string Id;
+
+            /// <summary>
+            ///     The name of this deployment.
+            /// </summary>
+            public readonly string Name;
+
+            /// <summary>
+            ///     The project that this deployment belongs to.
+            /// </summary>
+            public readonly string ProjectName;
+
+            /// <summary>
+            ///     The tags that are present on this deployment.
+            /// </summary>
+            public readonly List<string> Tags;
+
+            private readonly SpatialdManager spatiald;
+
+            internal LocalDeployment(SpatialdManager spatiald, string id, string name, string projectName, params string[] tags)
+            {
+                this.spatiald = spatiald;
+                Id = id;
+                Name = name;
+                ProjectName = projectName;
+                Tags = tags.ToList();
+            }
+
+            /// <summary>
+            ///     Adds the "dev_login" tag to this deployment asynchronously.
+            /// </summary>
+            /// <returns>A task which represents the underlying operation to add the tag.</returns>
+            /// <exception cref="InvalidOperationException">Thrown if the operation to set the tag fails.</exception>
+            public async Task AddDevLoginTag()
+            {
+                // TODO: Remove shim once tag functionality is added to `spot`: WF-1487 to track.
```

Comment:
Do we have a UTY ticket for this or is it expected that WF-1487 will be done before this PR is merged?

Reply:
Unsure when WF-1487 will be done, I'll raise a ticket and link it to the workflows one